ethereum/check_swap_creator_contract.go

```diff
@@ -104,6 +105,18 @@ func CheckForwarderContractCode(
 	ec *ethclient.Client,
 	contractAddr ethcommon.Address,
 ) error {
+	// mainnet override - since the forwarder contract deployed on mainnet is compiled
+	// with solidity 0.8.7, but we're using 0.8.19 for SwapCreator.sol, we can just
+	// check that the address is what's expected.
+	chainID, err := ec.ChainID(ctx)
+	if err != nil {
+		return err
+	}
+
+	if contractAddr == common.MainnetConfig().ForwarderAddr && chainID.Uint64() == common.MainnetChainID {
+		return nil
+	}
```

Comment:
Maybe not in this PR, but since the behavior can change depending on the compiler version, it might be worth figuring out a way to get the identically compiled contract in dev/stagenet.